### PR TITLE
Lower the stream timeout to detect control plan rotation faster.

### PR DESCRIPTION
**Pull request by perj**
We encountered a problem where fiaas-deploy-daemon would hang for an extended period during a control plan rotation.
A traceback indicated it was still listening for events from the previous control plane, that wasn't sending any. It's still uncertain if this was an active connection or one dropped silently by the server.

Lowering the stream_timeout is the chosen solution, but it comes with a few problems, especially that before this, the watcher thread for exit with an exception when the stream times out. That's solved by cathing the timeout and returning normally. Additionally, resource versions are now supported by the watching, including handling the 410 Gone response.

### CODE REVIEW

**Review by oyvindio — COMMENTED** [review]
I think this looks good overall 👍 I left some some minor comments in line

File: k8s/watcher.py [hunk 1]
```diff
@@ -48,11 +49,27 @@ def watch(self, namespace=None):
...
+        last_seen_resource_version = None
         while self._run_forever:
-            for event in self._model.watch_list(namespace=namespace):
-                o = event.object
-                key = (o.metadata.name, o.metadata.namespace)
-                if self._seen.get(key) == o.metadata.resourceVersion and event.type != WatchEvent.DELETED:
-                    continue
-                self._seen[key] = o.metadata.resourceVersion
-                yield event
+            try:
```
[the remark this answers is not in the document]
Comment (perj) [reply k8s/watcher.py:57]: watch_list now contains such a log.

File: k8s/config.py [hunk 2]
```diff
@@ -35,9 +35,20 @@
...
+#: * On some servers, it might take this long to detect a dropped connection. This speaks for a low value,
+#:   to detect the issue faster.
+#: * When connecting, a resourceVersion is used to resume, if still valid. This speaks for a low value,
+#:   to avoid them expiring.
+#: * During idle periods, there might not be any new resourceVersions.
+#:   Therefore a full quorum read each time a timeout is reached.
+#:   This speaks for a high value.
+#: 4.5 minutes is the default, set to detect the first case above in a reasonable time,
+#:while being just below the default resourceVersion expiration of 5 minutes.
+stream_timeout = 270
```
[the remark this answers is not in the document]
Comment (perj) [reply k8s/config.py]: Updated the docstring.

**Review by blockjesper — APPROVED** [review]

**Review by oyvindio — APPROVED** [review]
Looks good! In addition to handling ReadTimeoutError and adjusting the stream timeout, I think adding the watch bookmark handling should further improve the robustness of the Watcher. Nice work 👍

**Comment by oyvindio** [comment]
Closes #70

**Pull request merged**